src/EnergyPlus/IndoorIceRink.cc

```diff
@@ -0,0 +1,2797 @@
+                        ShowContinueError("The number of people has been reset to zero.");
+                        PeopleModifier = 0.0;
+                    }
+                } else { // no people schedule entered--assume that full number always present
+                    PeopleModifier = 1.0;
+                }
+            } else { // no heat gain schedule added--assume a zero value for Heat Gain per Person and no people present
+                HeatGainPerPerson = 0.0;
+                PeopleModifier = 0.0;
+            }
+            IRink(SysNum).PeopleHeatGain = PeopleModifier * HeatGainPerPerson * IRink(SysNum).MaxNumOfPeople;
+        }
+    }
+
+    void IceRinkFreezing(Real64 &FreezingLoad, // Freezing load
+                         int const SysNum,     // Index to refrigeration system
+                         int const SystemType  // Type of system (Direct of Indirect)
+    )
+    {
+        // Using/Aliasing
+        using FluidProperties::GetDensityGlycol;
+        using FluidProperties::GetSpecificHeatGlycol;
+        using ScheduleManager::GetCurrentScheduleValue;
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        static std::string const RoutineName("IceRinkFreezing");
+        Real64 SetPointTemp; // temperature "goal" for the radiant system [Celsius]
+        Real64 Length;       // Length of ice rink
+        Real64 Width;        // Width of ice rink
+        Real64 Depth;        // Thickness of ice
+        Real64 Volume;       // Volume of ice structure
+        Real64 CpWater;      // Specific heat of water
+        Real64 RhoWater;     // Density of water
+        Real64 QFusion(333550.00);
+        Real64 CpIce(2108.00);
+        Real64 QFreezing;
+        Real64 FloodWaterTemp;
+
+        if (SystemType == DirectSystem) {
+            SetPointTemp = DRink(SysNum).IceSetptTemp;
+            FloodWaterTemp = DRink(SysNum).FloodWaterTemp;
+            RhoWater = GetDensityGlycol(fluidNameWater, FloodWaterTemp, DRink(SysNum).WaterIndex, RoutineName);
+            CpWater = GetSpecificHeatGlycol(fluidNameWater, FloodWaterTemp, DRink(SysNum).WaterIndex, RoutineName);
+            Length = DRink(SysNum).LengthRink;
+            Width = DRink(SysNum).WidthRink;
+            Depth = DRink(SysNum).IceThickness;
+            Volume = Length * Width * Depth;
+        } else if (SystemType == IndirectSystem) {
+            SetPointTemp = IRink(SysNum).IceSetptTemp;
+            FloodWaterTemp = IRink(SysNum).FloodWaterTemp;
+            RhoWater = GetDensityGlycol(fluidNameWater, FloodWaterTemp, IRink(SysNum).WaterIndex, RoutineName);
+            CpWater = GetSpecificHeatGlycol(fluidNameWater, FloodWaterTemp, IRink(SysNum).WaterIndex, RoutineName);
+            Length = IRink(SysNum).LengthRink;
+            Width = IRink(SysNum).WidthRink;
+            Depth = IRink(SysNum).IceThickness;
+            Volume = Length * Width * Depth;
+        }
```

Comment:
These two blocks look *really* similar.  I think that splitting up the direct and indirect is leading to duplicated code and excessive effort on your part.  This should be considered moving forward because we don't want to end up with code that is difficult to maintain.